codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/AwsJson.kt

```diff
@@ -148,6 +149,7 @@ open class AwsJson(
             codegenContext,
             httpBindingResolver,
             ::awsJsonFieldName,
+            builderInstantiator = ClientBuilderInstantiator(codegenContext.symbolProvider),
```

Comment:
nit: I'd prefer it if we didn't default to the client implementation. Historically `codegen-server` depended on `codegen-client` and when I made changes, I defaulted to the client implementation in the core classes to not have to change the client code, since it came first.

But now that we've untangled things and the server and the client sit on equal footing, I think it's best that we don't default to any implementation in the `codegen-core` classes. There's still some old code that defaults to the client implementation, but new code should not follow that example.